### PR TITLE
Support type parameters

### LINKED ISSUE #1: Support type parameters correctly
It should be possible to support a type like this. We should be able to support `newtype`s and `data` types relatively easily.

```haskell
data Prefix a = Prefix
  { prefixFieldNameOne :: Maybe a
  , prefixFieldNameTwo :: Maybe a
  }

$(deriveHasFieldWith (dropPrefix "prefix") ''Prefix)
```

If you try this, you'll get

```
    • Expecting one more argument to 'Prefix'
      Expected a type, but 'Prefix' has kind '* -> *'
    • In the second argument of 'HasField', namely 'Prefix'
      In the instance declaration for
        'HasField "prefix" Prefix (Maybe a_a5flan)'
```

Because the instance we create doesn't contain the type parameter.

### PR DESCRIPTION
**Pull request by chiroptical**
Closes #1

### CODE REVIEW

File: src/DeriveHasField.hs [hunk 1]
```diff
@@ -63,3 +67,8 @@ lowerFirst :: String -> String
 lowerFirst = \case
   [] -> []
   (x : xs) -> toLower x : xs
+
+tyVarBndrToName :: TyVarBndr flag -> Name
+tyVarBndrToName = \case
+  PlainTV name _ -> name
+  KindedTV name _ _ -> name
```
Comment (chiroptical) [inline src/DeriveHasField.hs:74]: I have no idea what nuance exists here. I imagine most of the applications will be simple types like the one represented in the test.

**Review by evanrelf — APPROVED** [review]

File: src/DeriveHasField.hs (hunk 1 above)
Comment (lykahb) [reply src/DeriveHasField.hs:74]: It's a type variable with a kind annotation:

```
data OtherType a b (kinded :: *  -> *) (kindedSymbol :: Symbol) = OtherType
  { otherTypeField :: Maybe a
  , otherTypeOtherField :: Either a b
  , otherTypeKindedConstructor :: kinded ()
  , otherTypeKindedSymbol :: Proxy kindedSymbol
  }
```

Comment (lykahb) [reply src/DeriveHasField.hs:74]: The implementation probably works with the kinded type variables too - they should be inferred.

Comment (evanrelf) [reply src/DeriveHasField.hs:74]: I found that `KindedTV` is generated even for type parameters without annotations. Like when I was hacking on some TH code, I kept getting errors about `Foo (a :: Type) (b :: Type)` when I wrote the type as `data Foo a b = ...`.

Comment (chiroptical) [reply src/DeriveHasField.hs:74]: If I added `kindedSymbol` to `OtherType` the test it still works. I'll create an issue to add these to the test.

**Pull request merged**